bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -1903,7 +1904,7 @@ void ReplicaImp::onMessage<CheckpointMsg>(CheckpointMsg *msg) {
   if (msgIsStable && (msgSeqNum > lastExecutedSeqNum || msgEpochNum >= getSelfEpochNumber())) {
     auto pos = tableOfStableCheckpoints.find(msgGenReplicaId);
     if (pos == tableOfStableCheckpoints.end() || pos->second->seqNumber() <= msgSeqNum ||
```

Comment:
f ( (pos == tableOfStableCheckpoints.end() &&  msgEpochNum >= getSelfEpochNumber()) || 
       pos->second->seqNumber() <= msgSeqNum ||
        msgEpochNum >= pos->second->epochNumber()) {
}
//we don't want to add checkpoint message from previous epoch to tableOfStableCheckpoints